capstone-backend/src/test/java/com/google/sps/servlets/SearchServletTest.java

```diff
@@ -0,0 +1,122 @@
+package com.google.sps.servlets;
+ 
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.google.sps.data.VolumeData;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Collection;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import static org.mockito.Mockito.when;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class SearchServletTest {
+ 
+  @Mock
+  HttpServletRequest request;
+
+  @Mock
+  HttpServletResponse response;
+
+  private SearchServlet searchServlet;
+  private String jsonResponse;
+  private static final String artificialJsonResponse = "{\"items\": [{\"id\":\"ABCDEFG\",\"volumeInfo\":" +
+    "{\"title\":\"BookTitle\",\"authors\":[\"Name1\",\"Name2\"],\"imageLinks\":{\"thumbnail\":" +
+    "\"link\"}}},{\"id\":\"LMNOP\",\"volumeInfo\":{\"title\":\"BookTitle2\",\"authors\":" +
+    "[\"Name1\"],\"imageLinks\":{\"thumbnail\":\"Book2Link\"}}},{\"id\":\"PQRST\",\"volumeInfo\":" +
+    "{\"title\":\"BookTitle3\",\"authors\":[],\"imageLinks\":{\"thumbnail\":\"Book3Link\"}}}]}";
+
+  @Before
+  public void setup() {
+    searchServlet = new SearchServlet();
+    MockitoAnnotations.initMocks(this);
+    jsonResponse = populateServletResponse(0);
+  }
+
+  public String populateServletResponse(int numResults) {
```

Comment:
Really nice work exploring how to use Mockito to test Java servlets here! This work will help serve as an example for servlet testing with Mockito for future PR's!

Like we discussed offline, a potential next step here would be to try to mock out the response from the URL connection. Though you've done a great job here already writing tests that don't depend on the output from Google Books API - by writing "structural" tests that check the JSON format, # of books in response, etc.

Reply:
Thank you! It will be good to look into fully mocking those responses later